apps/user-ms/src/main/java/it/pagopa/selfcare/user/util/UserUtils.java

```diff
@@ -51,4 +46,26 @@ public static boolean checkIfNotFoundException(Throwable throwable) {
 
         return false;
     }
+    /**
+     * The filterProduct function takes in a UserInstitution object and an array of states.
+     * It then creates a list of OnboardedProductState objects from the array of strings, if the array is not null.
+     * If it is null, it sets relationshipStates to be null as well.
+     * Then, for each product in userInstitution's products list:
+     * if relationshipStates is not null and does not contain that product's status (which should be an OnboardedProductState),
+     * remove that product from the list.
+     */
+    public UserInstitution filterProduct(UserInstitution userInstitution, String[] states) {
+        List<OnboardedProductState> relationshipStates = Optional.ofNullable(states)
+                .map(this::convertStatesToRelationshipsState)
+                .orElse(null);
+
+        userInstitution.getProducts().removeIf(onboardedProduct -> !Objects.isNull(relationshipStates) && !relationshipStates.contains(onboardedProduct.getStatus()));
```

Comment:
you could check if userInstitution.getProducts() is not null....it is a potentially NullPointer